Slim/Utils/Scanner/Local.pm

```diff
@@ -315,6 +327,32 @@ sub rescan {
 
 		$class->updateTracks($dbh, \$changes, $paths, $next, $changedOnlyCount, $changedOnlySQL, $args);
 
+		# Completely rebuild contributor_album for all changed albums...
+		if ( $changedOnlyCount ) {
+			# ... first, now that tracks has been updated, add albums referenced now which weren't referenced before (in case the user has moved a track to a different album)
+			$log->error("Adding to temporary table for changed albums") unless main::SCANNER && $main::progress;
+			$dbh->do( qq{
+				INSERT INTO changed_albums
+				SELECT DISTINCT(tracks.album) AS album
+				FROM   changed
+				JOIN   tracks ON changed.url = tracks.url
+				WHERE NOT EXISTS (SELECT * FROM changed_albums WHERE changed_albums.album = tracks.album)
+			} );
```

Comment:
Wouldn't it be simpler/good enough to only populate this table here, rather than on line 314 _and_ here?

And would `INSERT OR IGNORE` be slower rather than the lookup on itself?

Reply:
Oh, I missed that there's not only the `deleteTracks()` call, but also `updateTracks()` in between them.

Reply:
Ok, the `INSERT OR IGNORE` would require some more tweaking of the code. But that looks "cleaner" to me:

```diff
diff --git a/Slim/Utils/Scanner/Local.pm b/Slim/Utils/Scanner/Local.pm
index 96b596f2e..2a290cf70 100644
--- a/Slim/Utils/Scanner/Local.pm
+++ b/Slim/Utils/Scanner/Local.pm
@@ -310,8 +310,9 @@ sub rescan {
 		if ( $changedOnlyCount ) {
 			$log->error("Build temporary table for changed albums") unless main::SCANNER && $main::progress;
 			$dbh->do('DROP TABLE IF EXISTS changed_albums');
+			$dbh->do("CREATE $createTemporary TABLE changed_albums (album INTEGER PRIMARY KEY UNIQUE)");
 			$dbh->do( qq{
-				CREATE $createTemporary TABLE changed_albums AS
+				INSERT INTO changed_albums
 					SELECT DISTINCT(tracks.album) AS album
 					FROM   changed
 					JOIN   tracks ON changed.url = tracks.url
@@ -332,11 +333,10 @@ sub rescan {
 			# ... first, now that tracks has been updated, add albums referenced now which weren't referenced before (in case the user has moved a track to a different album)
 			$log->error("Adding to temporary table for changed albums") unless main::SCANNER && $main::progress;
 			$dbh->do( qq{
-				INSERT INTO changed_albums
+				INSERT OR IGNORE INTO changed_albums
 					SELECT DISTINCT(tracks.album) AS album
 					FROM   changed
 						JOIN tracks ON changed.url = tracks.url
-				WHERE NOT EXISTS (SELECT * FROM changed_albums WHERE changed_albums.album = tracks.album)
 			} );
 			# ... now, rebuild contributor_album
 			$dbh->do( qq{
```